service_manager/android/src/main/java/com/baseflow/service_manager/EventStreamHandlerImpl.java

```diff
@@ -0,0 +1,57 @@
+package com.baseflow.service_manager;
+
+import android.bluetooth.BluetoothAdapter;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+
+import io.flutter.plugin.common.EventChannel;
+
+public class EventStreamHandlerImpl implements EventChannel.StreamHandler {
+
+    private final ServiceManager serviceManager;
+    private final Context applicationContext;
+
+    private BroadcastReceiver stateReceiver;
+    private EventChannel.EventSink eventSink;
+
+    public EventStreamHandlerImpl(ServiceManager serviceManager, Context applicationContext) {
+
+        this.serviceManager = serviceManager;
+        this.applicationContext = applicationContext;
+
+        stateReceiver = new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                if (eventSink == null) {
+                    return;
+                }
+
+                final int state = intent.getIntExtra(BluetoothAdapter.EXTRA_STATE, -1);
+                switch (state) {
+                    case BluetoothAdapter.STATE_OFF:
+                        eventSink.success(BluetoothAdapter.STATE_OFF);
```

Comment:
You just pass BluetoothAdapter.STATE_OFF, which is 10. However, the value you pass here to Flutter should be the same on all platforms, so you don't have to know on the flutter side from which platform it is coming.

I would make an enum in dart and java which have the same order and pass that enum (as int) to the other side.